components/match2/wikis/teamfortress/match_group_input_custom.lua

```diff
@@ -0,0 +1,385 @@
+---@return table
+function CustomMatchGroupInput.processPlayer(player)
+	return player
+end
+
+---@param data table
+---@param indexedScores table[]
+---@return table
+---@return table[]
+function CustomMatchGroupInput.getResultTypeAndWinner(data, indexedScores)
+	-- Map or Match wasn't played, set not played
+	if
+		Table.includes(NP_STATUSES, data.finished) or
+		Table.includes(NP_STATUSES, data.winner)
+	then
+		data.resulttype = 'np'
+		data.finished = true
+	-- Map or Match is marked as finished.
+	-- Calculate and set winner, resulttype, placements and walkover (if applicable for the outcome)
+	elseif Logic.readBool(data.finished) then
+		if MatchGroupInput.isDraw(indexedScores) then
+			data.winner = 0
+			data.resulttype = 'draw'
+			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 'draw')
+		elseif MatchGroupInput.hasSpecialStatus(indexedScores) then
+			data.winner = MatchGroupInput.getDefaultWinner(indexedScores)
+			data.resulttype = 'default'
+			if MatchGroupInput.hasForfeit(indexedScores) then
+				data.walkover = 'ff'
+			elseif MatchGroupInput.hasDisqualified(indexedScores) then
+				data.walkover = 'dq'
+			elseif MatchGroupInput.hasDefaultWinLoss(indexedScores) then
+				data.walkover = 'l'
+			end
+			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 'default')
+		else
+			local winner
+			indexedScores, winner = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
+			data.winner = data.winner or winner
+		end
+	end
+
+	--set it as finished if we have a winner
+	if Logic.isNotEmpty(data.winner) then
+		data.finished = true
+	end
+
+	return data, indexedScores
+end
+
+---@param opponents table[]
+---@param winner integer?
+---@param specialType string?
+---@param finished boolean|string?
+---@return table[]
+---@return integer?
+function CustomMatchGroupInput.setPlacement(opponents, winner, specialType, finished)
+	if specialType == 'draw' then
+		for key, _ in pairs(opponents) do
+			opponents[key].placement = 1
+		end
+	elseif specialType == 'default' then
+		for key, _ in pairs(opponents) do
+			if key == winner then
+				opponents[key].placement = 1
+			else
+				opponents[key].placement = 2
+			end
+		end
+	else
+		local temporaryScore
+		local temporaryPlace = -99
+		local counter = 0
+		for scoreIndex, opp in Table.iter.spairs(opponents, CustomMatchGroupInput.placementSortFunction) do
+			local score = tonumber(opp.score) or ''
+			counter = counter + 1
+			if counter == 1 and Logic.isEmpty(winner) then
+				if finished then
+					winner = scoreIndex
+				end
+			end
+			if temporaryScore == score then
+				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement) or temporaryPlace
+			else
+				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement) or counter
+				temporaryPlace = counter
+				temporaryScore = score
+			end
+		end
+	end
+
+	return opponents, winner
+end
```

Comment:
This is very likely existed prior to commons having one, so yeah I think this can be replaced over with the commons one. So that means Just kick this entire section?

Reply:
ok so removing this entire section didnt work, I do change  `CustomMatchGroupInput.`  by removing `Custom` already, not sure if this is where the leftside issue is causing